cardano-node/src/Cardano/Node/Configuration/Logging.hs

```diff
@@ -211,3 +232,56 @@ createLoggingLayer ver nodeConfig' = do
 
 shutdownLoggingLayer :: LoggingLayer -> IO ()
 shutdownLoggingLayer = shutdown . llSwitchboard
+
+-- The node provides the basic node's information for TraceForwarderBK.
+-- It will be sent once TraceForwarderBK is connected to an external process
+-- (for example, RTView).
+nodeBasicInfo :: NodeConfiguration
+              -> Consensus.TopLevelConfig (CardanoBlock c)
+              -> UTCTime
+              -> IO [LogObject Text]
+nodeBasicInfo nc cfg nodeStartTime' = do
+  let CardanoLedgerConfig cfgByron cfgShelley cfgAllegra cfgMary = Consensus.configLedger cfg
+      protocolDependentItems =
+        case ncProtocol nc of
+          ByronProtocol ->
+             let genesis = byronLedgerConfig cfgByron
+             in [ ("systemStartTime",  show (getSystemStart $ Consensus.configBlock cfg))
+                , ("slotLengthByron",  show (fromByronSlotLength $ genesisSlotLength genesis))
+                , ("epochLengthByron", show (fromByronEpochSlots $ Gen.configEpochSlots genesis))
+                ]
+          ShelleyProtocol ->
+            let genesis = shelleyLedgerGenesis $ shelleyLedgerConfig cfgShelley
+            in [ ("systemStartTime",          show (SL.sgSystemStart genesis))
+               , ("slotLengthShelley",        show (mkSlotLength $ SL.sgSlotLength genesis))
+               , ("epochLengthShelley",       show (SL.sgEpochLength genesis))
+               , ("slotsPerKESPeriodShelley", show (SL.sgSlotsPerKESPeriod genesis))
+               ]
+          CardanoProtocol ->
+            let genesisShelley = shelleyLedgerGenesis $ shelleyLedgerConfig cfgShelley
+                genesisAllegra = shelleyLedgerGenesis $ shelleyLedgerConfig cfgAllegra
+                genesisMary    = shelleyLedgerGenesis $ shelleyLedgerConfig cfgMary
+            in [ ("systemStartTime",          show (SL.sgSystemStart genesisShelley))
+               , ("slotLengthShelley",        show (mkSlotLength $ SL.sgSlotLength genesisShelley))
+               , ("slotLengthAllegra",        show (mkSlotLength $ SL.sgSlotLength genesisAllegra))
+               , ("slotLengthMary",           show (mkSlotLength $ SL.sgSlotLength genesisMary))
+               , ("epochLengthShelley",       show (SL.sgEpochLength genesisShelley))
+               , ("epochLengthAllegra",       show (SL.sgEpochLength genesisAllegra))
+               , ("epochLengthMary",          show (SL.sgEpochLength genesisMary))
+               , ("slotsPerKESPeriodShelley", show (SL.sgSlotsPerKESPeriod genesisShelley))
+               , ("slotsPerKESPeriodAllegra", show (SL.sgSlotsPerKESPeriod genesisAllegra))
+               , ("slotsPerKESPeriodMary",    show (SL.sgSlotsPerKESPeriod genesisMary))
```

Comment:
There's quite a bit of duplication here. Repeating all three for all the upcoming eras is not so nice. I would write a function that takes a `ShelleyGenesis era` and a `String` or `Text` ("Shelley", "Allegra", or "Mary").